hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/OnHeapMapScanP.java

```diff
@@ -0,0 +1,262 @@
+import com.hazelcast.jet.core.AbstractProcessor;
+import com.hazelcast.jet.core.Processor;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.jet.sql.impl.SimpleExpressionEvalContext;
+import com.hazelcast.map.impl.MapContainer;
+import com.hazelcast.map.impl.MapService;
+import com.hazelcast.partition.Partition;
+import com.hazelcast.spi.impl.NodeEngine;
+import com.hazelcast.sql.impl.exec.scan.KeyValueIterator;
+import com.hazelcast.sql.impl.exec.scan.MapScanExecIterator;
+import com.hazelcast.sql.impl.exec.scan.MapScanRow;
+import com.hazelcast.sql.impl.expression.Expression;
+import com.hazelcast.sql.impl.expression.predicate.TernaryLogic;
+import com.hazelcast.sql.impl.plan.node.JetMapScanMetadata;
+import com.hazelcast.sql.impl.row.HeapRow;
+
+import javax.annotation.Nonnull;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Function;
+
+import static com.hazelcast.jet.impl.util.Util.distributeObjects;
+import static com.hazelcast.jet.sql.impl.ExpressionUtil.evaluate;
+import static java.util.stream.Collectors.groupingBy;
+import static java.util.stream.Collectors.mapping;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * This is SQL internal processor to provide a backend for SQL operations (e.g., SELECT * FROM map).
+ * <p>
+ * The main idea here is to encapsulate all map walking process into IMapTraverser.
+ * There is a special {@link MapScanExecIterator} inside the traverser that allows
+ * it to walk over all map partitions and extract map entries.
+ * IMapTraverser does a setup of MapScanExecIterator and starts traversing map partitions.
+ * Simultaneously, usage of Traverser interface saves simplicity and cleanliness:
+ * it just returns all map entries which it can read.
+ */
+public class OnHeapMapScanP extends AbstractProcessor {
+    private static int DEFAULT_ORDINAL = -1;
+
+    protected IMapTraverser traverser;
+    protected Object[] pendingItems;
+    private final HazelcastInstance hazelcastInstance;
+    private final JetMapScanMetadata mapScanMetadata;
+    private final List<Integer> partitions;
+
+    public OnHeapMapScanP(
+            @Nonnull HazelcastInstance hazelcastInstance,
+            @Nonnull JetMapScanMetadata mapScanMetadata,
+            @Nonnull List<Integer> partitions
+    ) {
+        this.hazelcastInstance = hazelcastInstance;
+        this.mapScanMetadata = mapScanMetadata;
+        this.partitions = partitions;
+    }
+
+    @Override
+    protected void init(@Nonnull Context context) throws Exception {
+        NodeEngine nodeEngine = ((HazelcastInstanceImpl) hazelcastInstance).node.nodeEngine;
+
+        traverser = new IMapTraverser(
+                nodeEngine,
+                mapScanMetadata,
+                this.partitions
+        );
+        traverser.init(SimpleExpressionEvalContext.from(context));
```

Comment:
Why the `init()` is needed at all?

Reply:
AFAIU, I can take `SerializationService` only form `SimpleExpressionEvalContext`, which is `Context` interface implementation.

Reply:
But you create `traverser` just to immediately `init()` it.